### PR TITLE
Admin interface: buttons to show/hide action buttons

### PR DESCRIPTION
**Pull request by retostauffer**
Changes in the pkp-lib (show/hide action buttons).

### CODE REVIEW

**Comment by asmecher** [comment]
(@NateWr, I think I already assigned you the other PR for this one.)

**Comment by retostauffer** [comment]
Thanks for considering it, but this one might have some logic-issues and definitively needs some additional testing. But would be great to have a solution for the "Oh, a vivid button, let's press it (again)!" problem in the upcoming releases, it is one of our deal-breakers   :).

Thank you for for at least considering the idea, happy to help/discuss this issue if required (and if I'm able to do so). 

:1st_place_medal:

**Comment by NateWr** [comment]
Thanks @retostauffer! I'll definitely take a look at this. I'm a little bit behind on following up stuff like this, but if you don't hear from me this month please remind me!

**Comment by retostauffer** [comment]
`ping natewr` :). No pressure, no need to respond, just a reminder. I just could not fall asleep and this came into my mind. Greez - R

**Comment by NateWr** [comment]
Thanks for the reminder @retostauffer and I'm sorry for the delay. My queue is pretty backed up but I do want to get a look at this as it seems like a good idea. :+1:

**Comment by retostauffer** [comment]
Thanks, @NateWr ! No problem it took a bit longer, I will also need a bit to review the changes, for sure!  I'll put it onto my "to-do list [important]" and will come back to you as soon as possible.

Thanks, and a happy new year :).

File: pages/workflow/PKPWorkflowHandler.inc.php [hunk 1]
```diff
@@ -322,14 +323,85 @@ function editorDecisionActions($args, $request) {
 				__($action['title'])
 				);
 			}
+		} else {
+			$editorDecisions = array();
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply pages/workflow/PKPWorkflowHandler.inc.php]: Thanks, fixed in the new pull request.

File: pages/workflow/PKPWorkflowHandler.inc.php [hunk 2]
```diff
@@ -322,14 +323,85 @@ function editorDecisionActions($args, $request) {
...
+		}
+		
+		## pkel/retostauffer: search for newest/latest decision for the current stageID
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply pages/workflow/PKPWorkflowHandler.inc.php]: Removed.

File: pages/workflow/PKPWorkflowHandler.inc.php [hunk 3]
```diff
@@ -322,14 +323,85 @@ function editorDecisionActions($args, $request) {
...
+		
+		
+		// pkel/retostauffer: setting some labels to be able to manipulate frontend
+		$submissionStatus = (int) $submission->getStatus();
+		switch( $submissionStatus ) {
+			// If submission already declined
+			// Use: smarty key [...].Submission.declined
+			//      stageDecision set to NULL 
+			case STATUS_DECLINED:
+				$stage['label'] = 'Submission';
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply pages/workflow/PKPWorkflowHandler.inc.php]: No longer used, this way, output via smarty in the template files.

File: pages/workflow/PKPWorkflowHandler.inc.php [hunk 4]
```diff
@@ -322,14 +323,85 @@ function editorDecisionActions($args, $request) {
...
+			case STATUS_PUBLISHED:
+				$stage['label'] = 'Submission';
+				$stageDecision = "published";
+				break;
+			// Else our submission is queued (active or unassigned)
+			// Use: setting $stageDecision to current $submission->getStageId();
+			//      smarty key [...].Revision.<stageDecision> depending
+			//          on the current $submission->stageId();
+			case STATUS_QUEUED: # queued
+				$stageDecision = (int) $submission->getStageId();
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply pages/workflow/PKPWorkflowHandler.inc.php]: You are right, but I've revamped large parts of the logic here anyways.

File: pages/workflow/PKPWorkflowHandler.inc.php [hunk 5]
```diff
@@ -322,14 +323,85 @@ function editorDecisionActions($args, $request) {
...
+					default:
+						$stageDecision = "unknown";
+				}
+				break;
+			// Else unknown, not defined yet.
+			default:
+				$stageDecision = "unknown";
 		}
+		if(!empty($stageDecision)) {
+			$stageDecision = sprintf("editor.submission.workflowDecisions.stageDecision.%s.%s",
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply pages/workflow/PKPWorkflowHandler.inc.php]: Makes sense, no more dynamic key creation in the new version.

File: templates/workflow/editorialLinkActions.tpl [hunk 6]
```diff
@@ -24,12 +24,45 @@
...
 			{/if}
-			{foreach from=$editorActions item=action}
-				<li>
-					{include file="linkAction/linkAction.tpl" action=$action contextId=$contextId}
-				</li>
-			{/foreach}
 		</ul>
+
+		{* pkel/retostauffer: additional output, hide action buttons, and jquery functionality to show/hide action links *}
+		{if $isActiveStage}
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply templates/workflow/editorialLinkActions.tpl]: Good point, now shown everywhere (if buttons are shown).

File: templates/workflow/editorialLinkActions.tpl [hunk 7]
```diff
@@ -24,12 +24,45 @@
...
+					<p>
+						{translate key=$stageDecision}
+					</p><br>
+					<a href="#">{translate key="editor.submission.workflowDecisions.stageDecision.change"}</a>
+				{elseif ( !empty($stageDecision) )}
+					{translate key=$stageDecision}<br>
+					<a href="#">{translate key="editor.submission.workflowDecisions.stageDecision.change"}</a>
+				{/if}
+
+				<ul class="pkp_workflow_decisions {if (!empty($stageDecision) || ($submissionStatus == 4 || $submissionStatus == 3))}hide_buttons{/if}">
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply templates/workflow/editorialLinkActions.tpl]: Thanks, was not aware of this. Learned something new, but not required anymore.

File: templates/workflow/editorialLinkActions.tpl [hunk 8]
```diff
@@ -24,12 +24,45 @@
...
+					{* added pkel/retostauffer If a decision has been made for this state: hide the decision buttons. *}
+						<script>
```
[the remark this answers is not in the document]
Comment (retostauffer) [reply templates/workflow/editorialLinkActions.tpl]: Would be great, see comments on new pull request.

**Comment by retostauffer** [comment]
Hy @NateWr 

Unfortunately I had no idea how to re-use this pull request in combination with the new one. Thus, I opened a new one (https://github.com/pkp/pkp-lib/pull/4583). 

Left some comments here (in response to your wonderful comments), but after that this one could be closed I think. Thanks again! And sorry if I did the pull-requests in a wrong way, still some kind of magic to me.

**Pull request closed without merging**